### PR TITLE
Functional flashlight attachments

### PR DESCRIPTION
**Pull request by Humonitarian**
## About The Pull Request

Players can now attach flashlights to guns. Discussion requested for what weapons should allow this, in case if not all.

## Why It's Good For The Game

TactiCool

## Testing

I ran around the Eris with a Kadmin that had a flashlight on it

## Changelog
:cl:
add: Ironhammer flashlights can be installed on guns
/:cl:

### CODE REVIEW

**Comment by thevandie** [comment]
Yay, more good stiff from tg

**Comment by Humonitarian** [comment]
> Yay, more good stuff from tg

Excuse YOU, this has NO TG code in it whatsoever.
Everything is peak Bay and Eris quality code performing at its intended power level.

**Comment by ValoTheValo** [comment]
> Yay, more good stuff from tg

>TG
my brother in christ this is original code

**Comment by thevandie** [comment]
I meant the feature, not the code
Also, peak "bay" oxymoron lol, bay code is shit, there's a reason everyone wants to remove it at all costs

**Comment by Humonitarian** [comment]
> I meant the feature, not the code 

TG was not my source material when designing this feature.

> Also, peak "bay" oxymoron lol, bay code is shit, there's a reason everyone wants to remove it at all costs

Oh yeah? If bay is so bad how could I implement functional flashlight attachments?

**Comment by thevandie** [comment]
Because even the so-called shitty codebase of tg has them

**Comment by Humonitarian** [comment]
> Because even the so-called shitty codebase of tg has them

Who called TG shitty? They have some pretty good code quality, I even plan to port bot AI from them (significantly less laggier than M0RB-A)

**Review by TheShown911 — APPROVED** [review]

File: code/game/objects/items/weapons/tools/mods/_upgrades.dm [hunk 1]
```diff
@@ -333,6 +333,11 @@
 	if(weapon_upgrades[GUN_UPGRADE_BAYONET])
 		G.attack_verb = list("attacked", "slashed", "stabbed", "sliced", "torn", "ripped", "diced", "cut")
 		G.sharp = TRUE
+	if(weapon_upgrades[GUN_UPGRADE_FLASHLIGHT])
+		G.flashlight_attachment = TRUE
+		G.verbs += /obj/item/gun/proc/toggle_light
+	else
+		G.verbs -= /obj/item/gun/proc/toggle_light
```
Comment (Pink-Chink) [inline code/game/objects/items/weapons/tools/mods/_upgrades.dm:340]: This removes the flashlight ability when a mod is added after the flashlight.

Comment (Humonitarian) [reply code/game/objects/items/weapons/tools/mods/_upgrades.dm:340]: Oh.

**Review by hyperioo — CHANGES_REQUESTED** [review]
Fix for Pink Chink's comment?

**Review by hyperioo — APPROVED** [review]

**Pull request merged**